File: StringtoInteger(atoi).py
class Solution:
    def myAtoI(self, s: str):
        i = 0
        n = len(s)
        sign = 1
        result = 0

        while i< n and s[i] == " ":
            i += 1

        if i < n and (s[i] == "+" or s[i] == "-"):
            if s[i] == "-":
                sign = -1
            i += 1
        
        while i<n and s[i].isdigit():
            result = result * 10 + int(s[i])

            if sign == 1 and result > 2**31-1:
                return 2**31-1
            if sign == -1 and result > 2**31:
                return -2**31
            
            i += 1
        
        return sign * result
    
sol = Solution()
print(sol.myAtoI(" -32")) # Output: -32
            

File: StringtoInteger(atoi).js
const myAtoi = function (s) {
  let i = 0;
  let n = s.length;
  let sign = 1;
  let result = 0;

  while (i < n && s[i] === " ") {
    i++;
  }

  if (i < n && (s[i] === "+" || s[i] === "-")) {
    if (s[i] === "-") {
      sign = -1;
    }
    i++;
  }

  while (i < n && s[i] >= "0" && s[i] <= "9") {
    result = result * 10 + (s[i].charCodeAt(0) - "0".charCodeAt(0));

    if (sign === 1 && result > 2 ** 31 - 1) {
      return 2 ** 31 - 1;
    }
    if (sign === -1 && result > 2 ** 31) {
      return -(2 ** 31);
    }
    i++;
  }

  return sign * result;
};

console.log(myAtoi(" 42cba33")); // Output: 42